src/youtube.js

```diff
@@ -0,0 +1,1104 @@
+   * @returns {void}
+   * @private
+   * @public
+   */
+  static prepareVideoElement(): void {
+    Youtube._logger.debug('Prepare the video element for playing not supported');
+  }
+
+  /**
+   * The player playback rates.
+   * @type {Array<number>}
+   */
+  static PLAYBACK_RATES: Array<number> = [1];
+
+  /**
+   * @constructor
+   * @param {PKMediaSourceObject} source - The selected source object.
+   * @param {Object} config - The player configuration.
+   */
+  constructor(source: PKMediaSourceObject, config: Object) {
+    super();
+    this._eventManager = new EventManager();
+    this._init(source, config);
+  }
+
+  /**
+   * Restores the engine.
+   * @param {PKMediaSourceObject} source - The selected source object.
+   * @param {Object} config - The player configuration.
+   * @returns {void}
+   */
+  restore(source: PKMediaSourceObject, config: Object): void {
+    this.reset();
+    this._init(source, config);
+  }
+
+  /**
+   * shared reset of the engine.
+   * @returns {void}
+   * @private
+   */
+  _reset(): void {
+    this._stopSeekTargetWatchDog();
+    this._stopPlayingWatchDog();
+    this.detach();
+    if (this._api) {
+      this._api.stopVideo();
+    }
+    this._loaded = false;
+    this._firstPlaying = true;
+    this._isSeeking = false;
+    this._isAdaptiveBitrate = true;
+    this._playerTracks = [];
+    this._currentState = null;
+  }
+
+  /**
+   * Resets the engine.
+   * @returns {void}
+   */
+  reset(): void {
+    this._reset();
+    this._eventManager.removeAll();
+  }
+
+  /**
+   * Destroys the engine.
+   * @public
+   * @returns {void}
+   */
+  destroy(): void {
+    this._reset();
+    if (this._api) {
+      this._api.destroy();
+    }
+    this._eventManager.destroy();
+  }
+
+  /**
+   * Get the engine's id
+   * @public
+   * @returns {string} the engine's id
+   */
+  get id(): string {
+    return Youtube.id;
+  }
+
+  /**
+   * Listen to the video  events and triggers them from the engine.
+   * @public
+   * @returns {void}
+   */
+  attach(): void {}
+
+  /**
+   * Handles errors from the video element
+   * @returns {void}
+   * @private
+   */
+  _handleVideoError(): void {
+    this.dispatchEvent(new FakeEvent(EventType.ERROR, {}));
+  }
+
+  /**
+   * Remove the listeners of the video element events.
+   * @public
+   * @returns {void}
+   */
+  detach(): void {}
+
+  /**
+   * @returns {HTMLVideoElement} - The video element.
+   * @public
+   */
+  getVideoElement(): HTMLVideoElement {
+    return this._el;
+  }
+
+  /**
+   * Select a new video track.
+   * @param {VideoTrack} videoTrack - The video track object to set.
+   * @returns {void}
+   */
+  selectVideoTrack(videoTrack: VideoTrack): void {
+    if (this._api) {
+      if (videoTrack instanceof VideoTrack && (!videoTrack.active || this.isAdaptiveBitrateEnabled()) ) {
+        if (this.isAdaptiveBitrateEnabled()) {
+          this.dispatchEvent(EventType.ABR_MODE_CHANGED, {mode: 'manual'});
```

Comment:
not sure if this should be exported to a const / enum

Reply:
Good point. Added task to next sprint: FEC-8832 this change needs to be done across adapters and engines